src/NUnitFramework/framework/Internal/Reflect.cs

```diff
@@ -375,5 +375,107 @@ private static bool IsNullable(Type type)
                 && !type.GetTypeInfo().IsGenericTypeDefinition
                 && ReferenceEquals(type.GetGenericTypeDefinition(), typeof(Nullable<>));
         }
+
+        internal static IEnumerable<Type> TypeAndBaseTypes(this Type type)
+        {
+            for (; type != null; type = type.GetTypeInfo().BaseType)
+            {
+                yield return type;
+            }
+        }
+
+#if NETSTANDARD1_4
+        internal static Type GetInterface(this Type type, string name)
+        {
+            return type.GetTypeInfo().ImplementedInterfaces
+                .SingleOrDefault(implementedInterface => implementedInterface.FullName == name);
+        }
+#endif
+
+        /// <summary>
+        /// Same as <c>GetMethod(<paramref name="name"/>, <see cref="BindingFlags.Public"/> |
+        /// <see cref="BindingFlags.Instance"/>, <see langword="null"/>, <paramref name="parameterTypes"/>,
+        /// <see langword="null"/>)</c> except that it also chooses only non-generic methods.
+        /// Useful for avoiding the <see cref="AmbiguousMatchException"/> you can have with <c>GetMethod</c>.
+        /// </summary>
+        internal static MethodInfo GetNonGenericPublicInstanceMethod(this Type type, string name, Type[] parameterTypes)
+        {
+            for (var currentType = type; currentType != null; currentType = currentType.GetTypeInfo().BaseType)
```

Comment:
Minor, but this could be a `foreach(var currentType in type.TypeAndBaseTypes())` reusing your code from above. it might read better?

Reply:
Oh, I like that! Much better.